libs/wingsdk/src/target-sim/schedule.inflight.ts

```diff
@@ -36,19 +35,32 @@ export class Schedule
 
   // Calculate the delay for the next execution
   private nextDelay(interval: CronExpression) {
-    return interval.next().toDate().getTime() - Date.now();
+    const nextDate = interval.next().toDate();
+    this.context.addTrace({
+      type: TraceType.RESOURCE,
+      level: LogLevel.VERBOSE,
+      data: {
+        message: `Scheduling the next task to run at: ${nextDate} (the current time is ${new Date()})`,
+      },
+      sourcePath: this.context.resourcePath,
+      sourceType: SCHEDULE_FQN,
+      timestamp: new Date().toISOString(),
+    });
+    return nextDate.getTime() - new Date().getTime();
   }
 
   // Recursively schedule the function to be executed
   private scheduleFunction() {
+    const delay = this.nextDelay(this.interval);
     this.intervalTimeout = setTimeout(() => {
       this.runTasks();
       this.scheduleFunction();
-    }, this.nextDelay(this.interval));
+    }, delay);
   }
 
   public async init(context: ISimulatorContext): Promise<ScheduleAttributes> {
     this._context = context;
+    this.scheduleFunction();
```

Comment:
Is there a feasible test to make sure the trigger happens? Our schedule resource lacks granularity so I imagine there would be some pain

Reply:
We've got some end to end tests that test for e.g. the schedule activating every minute, I think here the issue was just with date calculation / subtraction.

There's _probably_ a way to check the behavior of date-based crons (that doesn't take hours to run) by writing a test where a cron is dynamically chosen during preflight based on the current time that `wing test` is executed, but it sounds like a headache in practice